source/game/field/obj/ObjectTwistedWay.cc

```diff
@@ -0,0 +1,355 @@
+}
+
+/// @addr{0x807FB8B0}
+bool ObjectTwistedWay::checkSpherePartialPushImpl(f32 radius, const EGG::Vector3f &v0,
+        const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut,
+        u32 timeOffset) {
+    return checkSphereImpl(radius, v0, v1, flags, pInfo, pFlagsOut, timeOffset, false, true);
+}
+
+/// @addr{0x807FBAC0}
+bool ObjectTwistedWay::checkSphereFullImpl(f32 radius, const EGG::Vector3f &v0,
+        const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut,
+        u32 timeOffset) {
+    return checkSphereImpl(radius, v0, v1, flags, pInfo, pFlagsOut, timeOffset, true, false);
+}
+
+/// @addr{0x807FBE6C}
+bool ObjectTwistedWay::checkSphereFullPushImpl(f32 radius, const EGG::Vector3f &v0,
+        const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut,
+        u32 timeOffset) {
+    return checkSphereImpl(radius, v0, v1, flags, pInfo, pFlagsOut, timeOffset, true, true);
+}
+
+/// @brief Helper function to calculate the wall's height offset in the wave for the wall collision
+/// check
+/// @param zAxisProgress how far along the object the sphere is on the z axis
+/// @param phase the current phase of the wave
+/// @return the height offset for the wall collision check
+f32 ObjectTwistedWay::calcWallHeightOffset(f32 zAxisProgress, s32 phase) {
+    f32 heightOffset =
+            EGG::Mathf::SinFIdx((phase * F_PI / PHASE_COUNT + WAVINESS * zAxisProgress) * RAD2FIDX);
+    heightOffset = AMPLITUDE *
+            ((zAxisProgress + 1.0f) *
+                    ((zAxisProgress + 1.0f) *
+                            ((zAxisProgress - 1.0f) * ((zAxisProgress - 1.0f) * heightOffset))));
+
+    return heightOffset;
+}
+
+bool ObjectTwistedWay::checkSphereWallCollision(f32 heightOffset, f32 radius, s32 phase,
+        const EGG::Vector3f &relativePos, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut, bool full,
+        bool push) {
+    auto [sinfidx, cosfidx] = EGG::Mathf::SinCosFIdx(heightOffset * RAD2FIDX);
+    f32 fVar3 = sinfidx * (relativePos.y - _B8 * 0.5f);
+    f32 dist = radius + (fVar3 - cosfidx * (relativePos.x + WIDTH));
+    EGG::Vector3f bbox = EGG::Vector3f(dist * cosfidx, dist * sinfidx, 0.0f);
+
+    if (dist <= 0.0f) {
+        dist = radius - (fVar3 - cosfidx * (relativePos.x - WIDTH));
+        bbox = EGG::Vector3f(-dist * cosfidx, dist * sinfidx, 0.0f);
+    }
+
+    if (dist > 0.0f) {
+        if (pInfo) {
+            pInfo->bbox.min = pInfo->bbox.min.minimize(bbox);
+            pInfo->bbox.max = pInfo->bbox.max.maximize(bbox);
+        }
+
+        if (pFlagsOut) {
+            if (push) {
+                CollisionDirector::Instance()->pushCollisionEntry(dist, pFlagsOut,
+                        KCL_TYPE_BIT(COL_TYPE_WALL), 0xc);
+            } else {
+                *pFlagsOut |= KCL_TYPE_BIT(COL_TYPE_WALL);
+            }
+        }
+
+        return true;
+    } else {
```

Comment:
Personal preference, but you don't need an `else` indentation if you have a guaranteed return in the if-statement above.

Reply:
Completely missed that, mb